lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -1089,6 +1090,30 @@ void SessionState::set_fsm_state(
   }
 }
 
+// Suspend the service due to all the remaining credits are transient.
+// Use the rg to trigger redirection
+bool SessionState::suspend_service_if_necesary_for_credit(
```

Comment:
I'm a little confused as to what this function does. So it checks all the charging credits in the session and if all of the existing credits are suspended, we put the credit in question into suspension?

If all of the credits in the session are already `suspended=true`, wouldn't the credit in question (the one being passed into the function) also be in suspension?

Reply:
Note there are 2 levels: `suspended=true` means the credit itself is suspended.  When the session is suspended that means *all* RGs are `suspended=true`

This function is run every time we set one RG into suspension just to check if we should suspend the whole session. If all the others RGs are suspended then is when we trigger the suspension, otherwise we do nothing